### PR TITLE
feat(console): theme toggle

### PR DESCRIPTION
**Pull request by polamoros**
Add a toggle to switch the console theme (light/dark/auto).

Three different options:
image
image
image

Hovering:
image

Added the filename to the tab
image

Added the filename to the header
image

resolves https://github.com/winglang/wing/issues/3116

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: apps/wing-console/console/ui/src/layout/vscode-layout.tsx [hunk 1]
```diff
@@ -48,12 +49,19 @@ export const VscodeLayout = ({ cloudAppState, wingVersion }: LayoutProps) => {
...
     logs,
     onResourceClick,
+    wingfile,
   } = useLayout({
     cloudAppState,
     defaultLogLevels: ["info", "warn", "error"],
   });
 
-  const { title } = useContext(AppContext);
+  const title = useMemo(() => {
```
Comment (ainvoner) [inline apps/wing-console/console/ui/src/layout/vscode-layout.tsx:58]: I think the use-layout should have this logic and return the actual value

File: apps/wing-console/console/ui/src/layout/vscode-layout.tsx [hunk 2]
```diff
@@ -48,12 +49,19 @@ export const VscodeLayout = ({ cloudAppState, wingVersion }: LayoutProps) => {
...
+    return `${wingfile.data ?? ""}`;
+  }, [wingfile.data]);
+
+  useEffect(() => {
+    document.title = title ? `${title} | Wing Console` : "Wing Console";
```
Comment (ainvoner) [inline apps/wing-console/console/ui/src/layout/vscode-layout.tsx:63]: haven't we discussed to follow vscode convention? fileName - Wing Console

**Review by ainvoner — CHANGES_REQUESTED** [review]

File: apps/wing-console/console/ui/src/layout/vscode-layout.tsx (hunk 2 above)
Comment (polamoros) [reply apps/wing-console/console/ui/src/layout/vscode-layout.tsx]: true, fixed

File: apps/wing-console/console/ui/src/layout/vscode-layout.tsx (hunk 1 above)
Comment (polamoros) [reply apps/wing-console/console/ui/src/layout/vscode-layout.tsx]: done

File: apps/wing-console/console/ui/src/layout/header.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            "rounded-3xl font-medium flex focus:outline-none",
+            "hover:bg-slate-200 hover:dark:bg-slate-600",
+            "transition-color duration-300 cursor-pointer",
+            "gap-x-1 px-1.5 py-0.5",
+          )}
+          onClick={toggleThemeMode}
+        >
+          {currentTheme === "light" && <SunIcon className="h-4" />}
+          {currentTheme === "dark" && <MoonIcon className="h-4" />}
+          <div className="font-light capitalize text-xs">{mode}</div>
```
Comment (ainvoner) [inline apps/wing-console/console/ui/src/layout/header.tsx:52]: maybe remove the strings for the regular dark/light modes, and only leave the Auto string?

**Review by ainvoner — CHANGES_REQUESTED** [review]

**Review by ainvoner — APPROVED** [review]

**Pull request merged**